src/requests.jl

```diff
@@ -248,25 +248,40 @@ for the FHIR `client`, and parse the JSON response with JSON3 as an object of ty
 
 See also [`request_json`](@ref) and [`request_raw`](@ref).
 """
-@inline function request(::Type{T},
-                         client::Client,
-                         verb::AbstractString,
-                         path::AbstractString;
-                         body = nothing,
-                         headers::AbstractDict = Dict{String, String}(),
-                         query::Union{AbstractDict, Nothing} = nothing,
-                         require_base_url::Symbol = :strict,
-                         kwargs...)::T where T
+@inline function request(
+    ::Type{T},
+    client::Client,
+    verb::AbstractString,
+    path::AbstractString;
+    body = nothing,
+    headers::AbstractDict = Dict{String, String}(),
+    query::Union{AbstractDict, Nothing} = nothing,
+    require_base_url::Symbol = :strict,
+    kwargs...,
+) where T
     _new_request_body = _write_struct_request_body(body)
-    response_body::String = request_raw(client,
-                                        verb,
-                                        path;
-                                        body = _new_request_body,
-                                        headers = headers,
-                                        query = query,
-                                        require_base_url = require_base_url)::String
-    response_object::T = JSON3.read(response_body,
-                                    T;
-                                    kwargs...)::T
+    response_body = request_raw(client,
+        verb,
+        path;
+        body = _new_request_body,
+        headers = headers,
+        query = query,
+        require_base_url = require_base_url,
+    )::String
+
+    # Recall that the default log levels are:
+    #     Error === LogLevel(2_000)
+    #     Warn === LogLevel(1_000)
+    #     Info === LogLevel(0)
+    #     Debug === LogLevel(-1_000)
+    #
+    # Ref: https://docs.julialang.org/en/v1/stdlib/Logging
+
+    # This is really only necessary if you run into errors when parsing the JSON.
+    @logmsg LogLevel(-5_000) "" response_body
+
+    @logmsg LogLevel(-3_000) "" JSON3.read(response_body)
```

Comment:
I'm not sure if this should be done, I am a bit worried that this will throw an error. For instance, this is problematic if the response is not in the JSON format. I also wonder what additional benefit this provides over just logging `response_body`?

Reply:
The default ConsoleLogger would catch the exception here, but in general, you're right that we shouldn't rely on that.

So let's explicitly make sure that this doesn't throw: #138 

The big benefit of logging this is that `JSON3.read(response_body)` will be pretty-printed to the logs; thus, when reading the logs, it's going to be a lot easier to read `JSON3.read(response_body)` than to read the raw `response_body`.

Reply:
Alright, I've merged #138 into this PR. Now, we will catch any exceptions that JSON3 might throw.